### DISCUSSION TITLE
torch

### DISCUSSION THREAD

**Discussion by ehartford** [tinygrad/tinygrad#6132] [General]
Do we really need to import torch?  It seems like a bad idea.

https://github.com/tinygrad/tinygrad/blob/d0513087e1df59e57d6957f7e23f0c4b506481b5/test/test_optim.py#L2

**Comment by ehartford** [comment]
I was in there trying to add GrokAdamW and I saw patterns that would force me to import the pytorch version of the optimizer (thus limiting us only to optimizers that are already implemented in pytorch?)

**Reply by ehartford** [reply]
It seems to me, we should not have any kind of dependency (even a test dependency) on pytorch

**Comment by chenyuxyz** [comment]
it does not need to be torch. fine as long as there's a reliable source to compare with. we have LARS and LAMB that compare with tonsorflow version because that's the reference for MLPerf

**Reply by ehartford** [reply]
I don't think we need to rely on an external implementation to compare - as opposed to first party test logic.  Why must we rely on an external implementation?  What if the optimizer is only implemented in tinygrad and no other third party frameworks?
